### ISSUE TITLE
In the table I don't find any selections in the dropdown field

### ISSUE THREAD

**Issue by rgleason** [registry/admiralty_tide_tables_pi#2]
See screenshot. Data needed. maybe it is xml?  I am guessing you have a data directory that the plugin refers to.  In windows we usually put this information under here:
C:\Program Files (x86)\OpenCPN 4.2.1\plugins\climatology_pi\data   (for example)

or if the user is going to be able to change the data and write to it, here
C:\Users\All Users\opencpn\plugins\weather_routing\data  (for example)
OR
C:\ProgramData\opencpn\plugins\weather_routing\data   (which is the same place)  (or maybe the particular user might be better)
I do not know how this is done in other plugins, you'll have to look at some examples of other plugins.

![admiralty-table](https://cloud.githubusercontent.com/assets/2363959/16451509/7c358356-3dd1-11e6-926b-1ad5c1764bd5.jpg)

**Comment by registry** [comment]
i'm currently working on loading/storing data.

**Comment by registry** [comment]
can you try out io under windows? i have added storing and loading (first shot).

After data is entered, one presses "Save" - it gets stored in internal container and is available in comboboxes
if one closes opencpn -> it goes on HDD
while starting opencpn -> is loaded automatically and is available in comboxes

UX is still to be improved.

**Comment by rgleason** [comment]
I'm sailing w/o my building computer. Will do so first chance on land. 3-4 days maybe.

**Comment by rgleason** [comment]
Tried compiling.  to get started I needed to remove ocpn_plugin.h  Then the first cmake -T v120_xp ..
worked.. The next one caused a bunch of errrors with wxwidgets it looks like..
I copied it all basically.
[admir-erros.txt](https://github.com/registry/admiralty_tide_tables_pi/files/364882/admir-erros.txt)

Looking forward to trying it out. Thanks

**Comment by registry** [comment]
ok. thnx.. will take look at this

**Comment by rgleason** [comment]
was this fixed?  Maybe I don't understand.  Are there no tide tables with this?  Are we supposed to build our own?

**Comment by registry** [comment]
No... im on bike ride in scotland... will be back like in one week.. then will have time for it.... 

Le 23 juillet 2016 20:50:06 GMT+01:00, Rick Gleason notifications@github.com a écrit :

> was this fixed?
> 
> ---
> 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/registry/admiralty_tide_tables_pi/issues/2#issuecomment-234736691

## 

Envoyé de mon téléphone Android avec K-9 Mail. Excusez la brièveté.

**Comment by rgleason** [comment]
Enjoy

**Comment by rgleason** [comment]
Someone was asking about English Tides and if there was a harmonic file.
This might help them.

**Comment by registry** [comment]
sorry for delay... i was sailing on north France to test this plugin...  in couple with REEDS it is indeed helpful

to clarify, there are no harmonic files... only data for standard and secondary ports to specific date/year published in reeds or ATT.... 

if one extends this plugin with sharing ability, data set can grow very rapidly with growing number of sharing users and in this way one will have to input less and less data manually... but this is future... 

now i try to implement basic functionality and remove glitches

**Comment by registry** [comment]
try now... i've fixed errors reported before... i hope

**Comment by rgleason** [comment]
Ok it compiles now, but there does not seem to be anything in the fields.
Are we missing a Data file or something?  Where should the Data file be stored? Where would I find it?
Perhaps it needs to be opened up? or maybe created?  

Please advise.
![nodata](https://cloud.githubusercontent.com/assets/2363959/17879893/6733731a-68c3-11e6-991f-cbe5caf2d964.jpg)

**Comment by registry** [comment]
Hi
this plugin in current form doesnt have any initial data.
Workflow is like follows:
1. By first execution of plugin user has nothing.
2. Then one is interested in tides of some standard port:
3. One looks up them in A.T.T or REEDS
3.1, if one wants or knows, that one will need them, one can select date and fill up the fields in "Standard Port" like it is standing in A.T.T or REEDS,
writing times like 05:34 as 0534 , 23:11 as 2311
3.2 -> by pressing "Save" entered data is stored in internal storage (initially in RAM -> writing ot disk while closing OpenCPN or Plugin)
4. while next start of plugin, previosly entered data is available for a given Date (!) in drop-down menu "Standard Port, Name"
5. If one wants to know tides of a secondary port, one looks up in A.T.T or REEDS its data and fills out for a given Year (!) fields "Secondary Port" again, writing times like 0023 or -0034 and mean tide differens as a normal double value like 0.45
5.1 -> Storing works like by "Standard Port"
6. To every secondary port one needs its corresponding standard port, which should be entered like in points 3 - 3.1- 3.2--4
7. Doesn one have both, secondary and standard port data for a given date, one selects them and presses "Calculate"
8. Vallues under "Secondary Port" are calculated accordin A.T.T. prescription.
9. One can correct data at anytime and save by "Save" 
10. Calculated data in NOT stored.

that is current state

there are possible extensions, like sharing data of standard and secondary ports. among users online. In this way with time one needs to enter less and less data. 
But i'm not sure if i will implement it, coz no time.

Hope that helps further

**Comment by skipperearly** [comment]
Has there been any progress on sharing data between users? How is it done?
It would be a shame to leave this usefull plugin so infantile.

**Comment by registry** [comment]
Hi. No, i do not sail now in English Channel and hence don't pursue development of this plugin. I can spend time on it only if need it to use.
As i've stated on OpenCPN forum, i would be glad to give over development of this plugin. 

Idea is to have a plugin, which allows to calculate Tides usint A.T.T. tables from UK using official way to calculate them. Having A.T.T. tables on hand as a book, one needs to perform relatively tedious calculation to get tides on secondary ports. This plugin should simplify this step. 
One enters data, which is published in A.T.T. and calculates tide for secondary ports.

For my 1 week trip it was helpful.

To make it useful for wider number of sailors, one needs to organize kind of "cloud", where once entered data from some user gets shared among everyone. One would need only to update database of standard ports and corrections for secondary ports. 

But effort to implement this is beyond my resources for this development. 
I would be happy to give over this plugin to community to let it grow and imiprove.
